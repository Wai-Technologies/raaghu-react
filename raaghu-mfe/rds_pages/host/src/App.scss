//@import "raaghu-react-themes/build/styles/default.css";
// Please dont remove below comment line Adding raaghu-theme in local side.
@import "../../../../raaghu-react-themes/src/styles/default.scss";

.main-body {
    margin-top: 29px;
    position: relative;
}
