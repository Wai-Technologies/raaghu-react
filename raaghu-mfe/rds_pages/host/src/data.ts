import { useTranslation } from "react-i18next";

export const getTranslatedThemeItems = () => {
  const { t } = useTranslation();

  const themeItems = [
    {
      id: 0,
      label: t("Light"),
      val: "Light",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/sun.json",
    },
    {
      id: 1,
      label: t("Dark"),
      val: "Dark",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/moon.json",
    },
    {
      id: 2,
      label: t("Semi Dark"),
      val: "Semi Dark",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/semi-dark.json",
    },
  ];

  return themeItems;
};

export const formsChildren = [
  {
    key: "5-0",
    label: "View",
    path: "/forms-view/:id",
    permission: "Forms.Form",
  },
];

export const getLabelForPath: any = (path: string, navItems: any) => {
  let label = null;
  for (const navItem of navItems) {
    if (navItem.path === path) {
      return navItem.label;
    }
    if (navItem.children) {
      label = getLabelForPath(path, navItem.children);
      if (label) {
        return label;
      }
    }
  }
  return label;
};