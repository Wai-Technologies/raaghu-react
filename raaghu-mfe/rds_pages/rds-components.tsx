export * from "../../raaghu-components/src/index";
