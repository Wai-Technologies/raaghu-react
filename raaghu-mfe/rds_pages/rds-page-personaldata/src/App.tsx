import React, { Suspense } from "react";
import PersonalData from "./personal-data/personal-data";

const App = () => {
    return (
        <Suspense>
            <PersonalData></PersonalData>
        </Suspense>
    );
};

export default App;