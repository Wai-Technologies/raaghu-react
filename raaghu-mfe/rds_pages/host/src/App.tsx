import React, { Suspense } from "react";
import "bootstrap/dist/js/bootstrap.bundle.min.js";
import "./App.scss";
import AuthHigherComponent from "./AuthHigherComponent";

const App = () => {
    return (
        <Suspense>
            <AuthHigherComponent></AuthHigherComponent>
        </Suspense>
    );
};

export default App;
