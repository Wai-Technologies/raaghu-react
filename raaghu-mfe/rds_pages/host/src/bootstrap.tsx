import App from "./App";
import ReactDOM from "react-dom/client";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../../libs/public.api";
import { Provider } from "react-redux";

(ReactDOM as any).createRoot(document.getElementById("root")).render(
    <Provider store={store}>
        <div>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </div>
    </Provider>
);

