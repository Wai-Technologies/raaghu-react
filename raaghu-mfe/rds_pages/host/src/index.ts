import("./bootstrap");
import "../../../web.config";
