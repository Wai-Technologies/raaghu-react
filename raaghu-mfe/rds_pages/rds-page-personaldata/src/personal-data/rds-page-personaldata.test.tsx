import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../../../libs/state-management";
import configureMockStore from "redux-mock-store";
import PersonalData from "./personal-data";
import { RdsButton, RdsIcon, RdsModal } from "../../../../../raaghu-elements/src";
import { RdsCompDatatable } from "../../../rds-components";

// Mock lodash-es directly in the test file
jest.mock("lodash-es", () => ({
  filter: jest.fn(),
  forEach: jest.fn(),
}));

global.location = {
    pathname: "../../../../libs/proxy/core",
  } as any;
  

jest.mock("react-router-dom", () => ({
  useNavigate: () => jest.fn(),
}));
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock("../../../../../raaghu-components/src/index.ts", () => ({
  ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
}));
jest.mock("../../../../../raaghu-elements/src/index.ts", () => ({
  ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
}));
jest.mock("../../../../libs/state-management/index.ts", () => ({
  store: {
    getState: jest.fn().mockReturnValue({
      persistedReducer: {
        configureStore: {
          configuration: {
            auth: {
              grantedPolicies: {},
            },
          },
        },
      },
    }),
    dispatch: jest.fn(),
    subscribe: jest.fn(),
  },
}));

// Mock the rds-elements and rds-components
jest.mock("../../../../../raaghu-elements/src", () => ({
  RdsButton: jest.fn(),
  RdsOffcanvas: jest.fn(),
  RdsAlert: jest.fn(),
  RdsSearch: jest.fn(),
  RdsModal: jest.fn(),
  RdsNavtabs: jest.fn(),
  RdsCheckbox: jest.fn(),
  RdsIcon: jest.fn(),
  RdsInput: jest.fn(),
  RdsDatePicker: jest.fn(),
}));

jest.mock("../../../../../raaghu-components/src", () => ({
  RdsCompApiScopeBasicResource: jest.fn(),
  RdsCompDatatable: jest.fn(),
  RdsCompAlertPopup: jest.fn(),
  RdsOffcanvas: jest.fn(),
  RdsCompProfilePicture: jest.fn(),
  RdsCompPersonalInfo: jest.fn(),
  RdsCompChangePassword: jest.fn(),
  RdsCompOrganizationTree: jest.fn(),
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

const mockStore = configureMockStore();

// Mock localStorage in your test setup
const localStorageMock: Storage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
  length: 0, // Add length property
  key: jest.fn(), // Add key method
};

global.localStorage = localStorageMock;
describe("Organization page", () => {
  it("renders organization page correctly", () => {
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    render(
      <Provider store={store}>
        <PersonalData />
      </Provider>
    );
    document.querySelector('[button="AbpGdpr.RequestPersonalData"]');
    document.querySelector('[button="AbpUi.Ok"]');
  });

it("Check personal data Page functions", () => {
  const tableHeaders = {
    displayName: "Display Name",
    key: "Key",
    datatype: "Datatype",
  };
  const personalTableData = {};
  const onActionSelection = jest.fn();
  const handlerRequestData = jest.fn();
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
   const data = render(
      <Provider store={store}>
        <PersonalData />
        <RdsModal modalId={"modal1"} children={undefined} />
        <RdsButton label="AbpGdpr.RequestPersonalData" onClick={handlerRequestData}/>
        <RdsIcon />
        <RdsButton label="AbpUi.Ok" databsdismiss="modal"/>
        <RdsCompDatatable tableHeaders={[tableHeaders]} tableData={[personalTableData]} pagination={true} onActionSelection={onActionSelection} recordsPerPage={10}/>
      </Provider>
    );
    expect(data).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
});
