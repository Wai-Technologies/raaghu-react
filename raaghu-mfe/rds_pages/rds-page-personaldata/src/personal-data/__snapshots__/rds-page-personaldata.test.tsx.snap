// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization page Check personal data Page functions 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid m-0 p-0 h-100"
      >
        <div
          class="card px-4 py-4 border-0 rounded-0 card-full-stretch"
        >
          <div
            class="row"
          >
            <div
              class="col-md-12 mb-3"
            >
              <div
                class="d-flex justify-content-end"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid m-0 p-0 h-100"
    >
      <div
        class="card px-4 py-4 border-0 rounded-0 card-full-stretch"
      >
        <div
          class="row"
        >
          <div
            class="col-md-12 mb-3"
          >
            <div
              class="d-flex justify-content-end"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
